helix-view/src/document.rs

```diff
@@ -638,6 +639,10 @@ impl Document {
         Ok(())
     }
 
+    pub fn get_path(&self) -> Option<PathBuf> {
+        self.path.clone()
+    }
```

Comment:
It's better to let the caller decide when/if to clone:

```suggestion
    pub fn path(&self) -> Option<&PathBuf> {
        self.path.as_ref()
    }
```

Then `Env` should use `doc.path().cloned()`.